File: sixup.py

import pandas as pd
import numpy as np
import matplotlib
import matplotlib.pyplot as plt

matplotlib.use('agg')

from sklearn.linear_model import LinearRegression

val =  pd.read_csv('sixup.csv')
col_names = list(val.columns.values)
clf = LinearRegression()
input_name = raw_input('enter which value you want to plot, (2-9): ')
Y = np.array(val['Max of Federal Loan 3-Year Default Rate'])
X = np.array(val.icol(int(input_name)))
X = X[:,np.newaxis]
clf.fit(X,Y)


X1 = []
Y1 = []
N = len(X)
for i in range(0,N):
    if (X[i] != 0 and Y[i] != 0):
        
        X1.append(X[i])
        Y1.append(Y[i])
    else:
        X1.append(0)
        Y1.append(0)
        
X1= np.array(X1)       
X1= X1[:,np.newaxis]
plt.plot(X1,Y1,'k.')
plt.plot(X1, clf.predict(X1), color='red')

plt.legend(loc=2)
plt.xlabel(col_names[1])
plt.ylabel(col_names[int(input_name)])
plt.title("%s versus %s" % (col_names[1],col_names[int(input_name)]))

xarray = np.array([i*0.001 for i in range(0,N)])
xarray = xarray[:,np.newaxis]

difference = clf.predict(xarray) - Y1

plt.grid()

#PLOT
fig1 = plt.figure(1)
#Plot Data-model
#Residual plot
frame2=fig1.add_axes((0.15,-.3,.76,.2))        
plt.plot(xarray,difference,'--')
plt.title('Residuals')

plt.grid()
plt.show()
print 'rsquared:',clf.score(X,Y), 'coefficients',clf.coef_

File: sixup.m
%%Christian Hansen
%%Matlab R2010a
%%Wednesday, August 26th, 2015

%%This code will explore several parameters effects upon the 3-year default
%%rate, by plotting these values against the 3-year default rate.


clear
M = importdata('sixup.dat');
%%import data as .dat so that I can use it properly and remove the grand
%%total from the bottom of the csv made from the pivot table file so not to misrepresent the data 
%%look at data from the .dat. used max values because average brought up
%%division by zero.

%%list of column names as stored in the .dat file and .csv
colnames = {'Max of Federal Loan 3-Year Default Rate';...
    'Max of 2012 4-Year Grad Rate';...
    'Max of % Pell Recipients Among Freshmen';...
    'Max of Estimated Median SAT / ACT';...
    'Max of Average Student Loan (all sources)';...
    'Max of Average Net Price After Grants';...
    'Max of % Degrees Awarded in Science, Technology, Engineering, and Math';...
    'Max of % Admitted';...
    'Max of Average High School GPA Among College Freshmen'};

%%getting rid of zeros which were what was from the original
%% data set marked either non-value (-) or ds. The pivot table
%% automatically changed these values.
N= 2229; % row count

input_value = input('which values to plot (2-9): ');

figure();
for i=1:2229
    %%didn't work as expected, but the range can be adjusted manually to
    %%see trends, given further time I'd spend my efforts working out the
    %%filtering as that should be a key step.
    
if M(i,1) > 0 && M(i,input_value) > 0
     m1(i) = M(i,1);
     m2(i) = M(i,input_value);
% else
%     m1(i)=10;
%     m2(i) = 10;
end
end

%%I tried to do fitting with regression, but ended up 
%%simply using the basic fitting capabilities.

plot(m2,m1,'k.')
xlabel(colnames(input_value))
ylabel(colnames(1))
lab1=char(colnames(1));
lab2=char(colnames(input_value));

str=sprintf('%s versus %s',lab1,lab2);
title(str)